### PR TITLE
Remove sinon dependency and fix requestAnimationFrame mocks

### PR DESCRIPTION
**Pull request by ecraig12345**
#### Pull request checklist

- [x] Include a change request file using `$ yarn change`

#### Description of changes

Remove dependency on `sinon` since we rarely use it, and the usage can be replaced with jest mocks. This was partly motivated because sinon is one of the things with a postinstall script--though now that I look at it, turns out it just [asks for funding](https://github.com/sinonjs/sinon/blob/v4.5.0/scripts/support-sinon.js) rather than downloading stuff, but still doesn't hurt to remove.

The one potentially breaking part here is that under `@uifabric/utilities/lib/test` there's a helper `setRenderSpy` which relies on sinon. We don't use that anywhere in our repo, and it's not technically part of the public API since it's not exported from the package root.

While testing this, I ran into issues with `safeRequestAnimationFrame.test.tsx` which made me realize that the `requestAnimationFrame` mock all our packages were using was broken (didn't return the timeout ID and didn't have a corresponding `cancelAnimationFrame`). So I added proper mocks in `scripts/jest/jest-setup.js` and deleted the incorrect mocks from all packages.

### CODE REVIEW

File: packages/react-next/src/components/OverflowSet/OverflowSet.test.tsx [hunk 1]
```diff
@@ -33,38 +32,26 @@ function getPersistedKeytip(keytipManager: KeytipManager, keySequences: string[]
   return ktp ? ktp.keytip : undefined;
 }
 
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const noOp = (): any => undefined;
```
Comment (xugao) [inline packages/react-next/src/components/OverflowSet/OverflowSet.test.tsx:36]: so many `noOp` implemented everywhere :) perhaps move this to our test utils? 
Don't have to be in this PR

File: packages/utilities/src/test/setRenderSpy.ts [hunk 2]
```diff
@@ -1,10 +1,17 @@
...
+// @ts-ignore
 import * as sinon from 'sinon';
 
 /**
  * Replaces the enzyme render function with a spy that can be used to count how many
  * times the render function was called in a test.
  * @param wrapper - The wrapper around the component instance for which you want to spy upon
+ * @deprecated This package no longer uses sinon. Please copy this function locally to continue using it.
  */
 export function setRenderSpy(wrapper: ReactWrapper<{}, {}>): sinon.SinonSpy {
```
Comment (xugao) [inline packages/utilities/src/test/setRenderSpy.ts:16]: I'd say let's just remove this. This was never officially supported from `utilities` package

**Review by xugao — APPROVED** [review]

**Review by khmakoto — APPROVED** [review]

File: packages/office-ui-fabric-react/src/components/OverflowSet/OverflowSet.test.tsx [hunk 3]
```diff
@@ -32,11 +31,13 @@ function getPersistedKeytip(keytipManager: KeytipManager, keySequences: string[]
...
 }
 
+const noOp = (): any => undefined;
+
 describe('OverflowSet', () => {
   describe('snapshot tests', () => {
     test('basicSnapshot', () => {
-      const onRenderItem = sinon.spy();
-      const onRenderOverflowButton = sinon.spy();
+      const onRenderItem = jest.fn();
```
[the remark this answers is not in the document]
Comment (ecraig12345) [reply packages/office-ui-fabric-react/src/components/OverflowSet/OverflowSet.test.tsx:39]: It's a required prop. Though I did miss changing this instance to the general `noOp`.

File: packages/react-next/src/components/OverflowSet/OverflowSet.test.tsx (hunk 1 above)
Comment (ecraig12345) [reply packages/react-next/src/components/OverflowSet/OverflowSet.test.tsx:36]: I think it's a simple enough thing that I'd rather just declare it in each file as needed. Kind of like how you probably don't need the `leftpad` npm package. :P

File: packages/utilities/src/test/setRenderSpy.ts (hunk 2 above)
Comment (ecraig12345) [reply packages/utilities/src/test/setRenderSpy.ts:16]: Sounds good

**Comment by layershifter** [comment]
@ecraig12345 can be merge conflicts solved there? Do you need any help with it? ⛑️

**Review by xugao — APPROVED** [review]

**Pull request merged**